File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int currIndex = 0;

        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] != nums[currIndex]) {
                currIndex++;
                nums[currIndex] = nums[i];
            }
        }

        return currIndex + 1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int currNum = nums[0];
        int unique = 1;

        for (int i = 1; i < nums.length; i++) {
            if (currNum != nums[i]) {
                currNum = nums[i];
                nums[unique] = currNum;
                unique++;
            }
        }

        return unique;
    }
}